core/src/net/udp_socket.cpp

```diff
@@ -0,0 +1,186 @@
+#include <cstring>
+
+#ifdef _WIN32
+#include <ws2tcpip.h>
+#else
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#endif
+
+#include <cubos/core/log.hpp>
+#include <cubos/core/net/udp_socket.hpp>
+#include <cubos/core/reflection/external/cstring.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+#include <cubos/core/reflection/external/string.hpp>
+
+using cubos::core::net::UdpSocket;
+
+UdpSocket::UdpSocket()
+{
+#ifdef _WIN32
+    WSADATA wsa;
+    CUBOS_ASSERT(WSAStartup(MAKEWORD(2, 2), &wsa) == 0, " WSAStartup failed: {}", WSAGetLastError());
+#endif
+}
+
+UdpSocket::~UdpSocket()
+{
+    unbind();
+}
+
+/// @brief Cross-platform utility to get last error message from stderr.
+/// @return Message.
+/// @todo Move this to a higher module for general purposes?
+static std::string error()
+{
+#ifdef _WIN32
+    static char message[256] = {0};
+    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, 0, WSAGetLastError(), 0, message, 256, 0);
+    char* nl = strrchr(message, '\n');
+    if (nl)
+        *nl = 0;
+    return {message};
+#else
+    return {strerror(errno)};
+#endif
+}
+
+bool UdpSocket::bind(uint16_t port, const Address& address)
+{
+    mSock = socket(AF_INET, SOCK_DGRAM, 0);
+    if (mSock == INVALID_SOCKET)
+    {
+        CUBOS_ERROR("Failed to allocate UDP socket: {}", error());
+        return false;
+    }
```

Comment:
This should be done on the constructor, as for clients, you don't need to call `bind`, you can just `send`.